### ISSUE TITLE
Setting authentication?

### ISSUE THREAD

**Issue by mepatterson** [allmarkedup/lookbook#74]
We have components that expect to be passed the current authenticated user. As an engine, it seems like Lookbook doesn't care about the app's ApplicationController, so the `current_user` is not available. Is there a way to provide lookbook preview classes with the user that is currently signed-in to the main app?

**Comment by allmarkedup** [comment]
Hey @mepatterson. So Lookbook doesn't actually have its own preview controller, it just reuses the ViewComponent preview controller to render the components. You can determine which controller to use via the [ViewComponent config](https://viewcomponent.org/guide/previews.html#configuring-preview-controller) - for example:

```ruby
config.view_component.preview_controller = "MyPreviewController"
```

This controller should extend the base `ViewComponentsController` that VC provides - the Lookbook demo app has [an example of this](https://github.com/allmarkedup/lookbook-demo/blob/main/app/controllers/preview_controller.rb).

Hope that helps! If you have any issues getting it working just let me know.

**Comment by mepatterson** [comment]
Yeah so I tried exactly that. I created a new controller that inherits from the view component controller and added a before_action where I was going to set up the auth. It worked if I visited the default VC preview url, but when visiting lookbook it seemed to ignore that new controller. Any ideas what I was doing wrong?

**Comment by allmarkedup** [comment]
Ah ok - if it works when using the VC preview url then that sounds like a bug, not something you are doing wrong. It should be no different. 

I'll investigate as soon as I have a chance and update you once it's fixed. Thanks for the report 👍

**Comment by allmarkedup** [comment]
Hi again @mepatterson. I believe I've tracked down the bug and have included a fix in the latest beta release, `v0.7.2.beta.4`.

Could you give that a try whenever you have a chance and let me know if it resolves your problem?

**Comment by allmarkedup** [comment]
Great, I'll close this for now then - will hopefully get the 'proper' release out later this week 👍
